website/public/js/controllers/groupsCtrl.js

```diff
@@ -229,6 +235,11 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
   .controller('ChatCtrl', ['$scope', 'Groups', 'User', '$http', 'ApiUrl', 'Notification', 'Members', '$rootScope', function($scope, Groups, User, $http, ApiUrl, Notification, Members, $rootScope){
     $scope.message = {content:''};
     $scope._sending = false;
+
+    $scope.preview = false;
+    $scope.togglePreview = function(){
+      $scope.preview = !$scope.preview;
+    }
```

Comment:
You could move this part of the preview functionality into a directive. That way it could be added to any text box regardless of the controller in charge for that page/section.